zulipterminal/ui_tools/boxes.py

```diff
@@ -132,18 +130,17 @@ def autocomplete_mentions(self, text: str, prefix_string: str
 
     def autocomplete_streams(self, text: str) -> List[str]:
         streams_list = self.view.pinned_streams + self.view.unpinned_streams
-        stream_names = [stream[0]
-                        for stream in streams_list
-                        if match_stream(stream, text[1:])]
+
+        streams = filtered_typeahead_names(streams_list, text, match_stream, 1)
+        stream_names = [stream[0] for stream in streams]
```

Comment:
This change has two list comprehensions, one inside a function, compared to just one list comprehension. That doesn't make it necessarily unwise, but if this can be seen more clearly worthwhile in the context of the other PR, let's put it there?

Reply:
Done :+1: